torch/serialization.py

```diff
@@ -203,16 +203,39 @@ def persistent_id(obj):
 def load(f, map_location=None, pickle_module=pickle):
     """Loads an object saved with :func:`torch.save` from a file.
 
-    torch.load can dynamically remap storages to be loaded on a different device
-    using the map_location argument. If it's a callable, it will be called with
-    two arguments: storage and location tag. It's expected to either return a
-    storage that's been moved to a different location, or None (and the location
-    will be resolved using the default method). If this argument is a dict it's
-    expected to be a mapping from location tags used in a file, to location
-    tags of the current system.
-
-    By default the location tags are 'cpu' for host tensors and 'cuda:device_id'
-    (e.g. 'cuda:2') for cuda tensors. User extensions can register their own
+    torch.load uses python's normal unpickling facilities but treats storages,
+    which underlie tensors, specially.  Storages are first deserialized on the
+    cpu.  Then, by default, torch.load tries to move the storage to the device
+    it was saved from, raising an exception if this fails because the device
+    does not exist.  If you serialize a cuda tensor with torch.save, when you
+    deserialize it with torch.load, it will be created on the device it was
+    originally on from a storage that has been moved there.
+
+    But torch.load can also dynamically remap tensors to be loaded on a
+    different device using the map_location argument.  To understand how this
+    works, we have to understand how torch.load knows what device to try to
+    move each storage to.  It knows this because torch.save saves a location
+    tag with each storage that says what device it was on.  The builtin
+    location tags are 'cpu' for host tensors and 'cuda:device_id'
+    (e.g. 'cuda:2') for cuda tensors.
```

Comment:
I don't feel that this blogpost-like style is good for the docs. Let's try to keep them brief. Second sentence doesn't add any meaning. Just say sth like "torch.load can also dynamically remap tensors to a different device, using the map_location argument. Each serialized storage has a device tag associated with it, which uniquely identifies the device it was on. The builtin location tags are ..."

Reply:
It seems to me if we just cut the second sentence, then it is strange to suddenly be talking about location tags.  Why are they important.  So, since you don't like the second sentence, I am moving discussion of location tags into the next paragraph, where it can be properly motivated.